### PR TITLE
Permissions: flush cache, don't use an in-memory cache

### PR DESCRIPTION
**Pull request by camhashemi**
@andrewconner @ryanpbrewster

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/OrganizationCommander.scala [hunk 1]
```diff
@@ -468,6 +471,11 @@ class OrganizationCommanderImpl @Inject() (
   def unsafeSetAccountFeatureSettings(orgId: Id[Organization], settings: OrganizationSettings)(implicit session: RWSession): OrganizationSettingsResponse = {
     val currentConfig = orgConfigRepo.getByOrgId(orgId)
     val newConfig = orgConfigRepo.save(currentConfig.withSettings(settings))
+    if (currentConfig.settings != settings) {
+      SafeFuture {
+        orgMembershipRepo.getAllByOrgId(orgId).foreach(mem => eliza.flush(mem.userId))
+      }
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/OrganizationCommander.scala]: future inside of db session will blow up

File: kifi-backend/modules/shoebox/app/com/keepit/common/cache/ShoeboxCacheModule.scala [hunk 2]
```diff
@@ -427,7 +427,7 @@ case class ShoeboxCacheModule(cachePluginModules: CachePluginModule*) extends Ca
 
   @Provides @Singleton
   def orgPermissionsNamespaceCache(stats: CacheStatistics, accessLog: AccessLog, innerRepo: InMemoryCachePlugin, outerRepo: FortyTwoCachePlugin) =
-    new OrganizationPermissionsNamespaceCache(stats, accessLog, (innerRepo, 5 minutes), (outerRepo, 7 days))
+    new OrganizationPermissionsNamespaceCache(stats, accessLog, (outerRepo, 7 days))
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/common/cache/ShoeboxCacheModule.scala:430]: :+1: good catch

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/OrganizationCommander.scala (hunk 1 above)
Comment (camhashemi) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/OrganizationCommander.scala]: :+1: good catch

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/OrganizationCommander.scala [hunk 3]
```diff
@@ -374,6 +376,19 @@ class OrganizationCommanderImpl @Inject() (
...
+  def unsafeSetAccountFeatureSettings(orgId: Id[Organization], settings: OrganizationSettings)(implicit session: RWSession): OrganizationSettingsResponse = {
+    val currentConfig = orgConfigRepo.getByOrgId(orgId)
+    val newConfig = orgConfigRepo.save(currentConfig.withSettings(settings))
+
+    val members = orgMembershipRepo.getAllByOrgId(orgId)
+    if (currentConfig.settings != settings) {
+      session.onTransactionSuccess {
+        members.foreach(mem => eliza.flush(mem.userId))
+      }
+    }
```
Comment (camhashemi) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/OrganizationCommander.scala:388]: @ryanpbrewster

**Pull request merged**